app/models/secret_storage.rb

```diff
@@ -50,6 +50,43 @@ def self.keys
     end
   end
 
+  class HashicorpVault
+
+    SECRET_BACKEND ='secret/'.freeze
+    # we don't really want other directories in here,
+    # and there may be other chars that we find we don't like
+    ENCODINGS = {"/": "%2F"}
+
+    def self.read(key)
+      result = Vault.logical.read(SECRET_BACKEND  + self.encode(key))
+      result.data[:secret_data]
+    end
+
+    def self.write(key, data)
+      Vault.logical.write(SECRET_BACKEND + self.encode(key), secret_data: data)
+    end
+
+    def self.delete(key)
+      Vault.logical.delete(SECRET_BACKEND + key)
```

Comment:
using the same code 3 times `SECRET_BACKEND + key` -> extract into a method ... will also fix the bug of delete not being able to remove with slashes ...